lib/jsdom/living/interfaces.js

```diff
@@ -1,186 +1,187 @@
-  require("./generated/HTMLTextAreaElement.js"),
-  require("./generated/HTMLUnknownElement.js"),
-  require("./generated/HTMLFrameElement.js"),
-  require("./generated/HTMLFrameSetElement.js"),
-  require("./generated/HTMLIFrameElement.js"),
-  require("./generated/HTMLEmbedElement.js"),
-  require("./generated/HTMLObjectElement.js"),
-  require("./generated/HTMLParamElement.js"),
-  require("./generated/HTMLVideoElement.js"),
-  require("./generated/HTMLAudioElement.js"),
-  require("./generated/HTMLTrackElement.js"),
-
-  require("./generated/SVGElement.js"),
-  require("./generated/SVGGraphicsElement.js"),
-  require("./generated/SVGSVGElement.js"),
-  require("./generated/SVGTitleElement.js"),
-  require("./generated/SVGAnimatedString"),
-  require("./generated/SVGNumber"),
-  require("./generated/SVGStringList"),
-
-  require("./generated/Event"),
-  require("./generated/CloseEvent"),
-  require("./generated/CustomEvent"),
-  require("./generated/MessageEvent"),
-  require("./generated/ErrorEvent"),
-  require("./generated/HashChangeEvent"),
-  require("./generated/PopStateEvent"),
-  require("./generated/StorageEvent"),
-  require("./generated/ProgressEvent"),
-  require("./generated/PageTransitionEvent"),
-
-  require("./generated/UIEvent"),
-  require("./generated/FocusEvent"),
-  require("./generated/InputEvent"),
-  require("./generated/MouseEvent"),
-  require("./generated/KeyboardEvent"),
-  require("./generated/TouchEvent"),
-  require("./generated/CompositionEvent"),
-  require("./generated/WheelEvent"),
-
-  require("./generated/BarProp"),
-  require("./generated/External"),
-  require("./generated/Location"),
-  require("./generated/History"),
-  require("./generated/Screen"),
-  require("./generated/Performance"),
-  require("./generated/Navigator"),
-
-  require("./generated/PluginArray"),
-  require("./generated/MimeTypeArray"),
-  require("./generated/Plugin"),
-  require("./generated/MimeType"),
-
-  require("./generated/FileReader"),
-  require("./generated/Blob"),
-  require("./generated/File"),
-  require("./generated/FileList"),
-  require("./generated/ValidityState"),
-
-  require("./generated/DOMParser"),
-  require("./generated/XMLSerializer"),
-
-  require("./generated/FormData"),
-  require("./generated/XMLHttpRequestEventTarget"),
-  require("./generated/XMLHttpRequestUpload"),
-  require("./generated/XMLHttpRequest"),
-  require("./generated/WebSocket"),
-
-  require("./generated/NodeIterator"),
-  require("./generated/TreeWalker"),
-
-  require("./generated/AbstractRange"),
-  require("./generated/Range"),
-  require("./generated/StaticRange"),
-  require("./generated/Selection"),
-
-  require("./generated/Storage"),
-
-  require("./generated/ShadowRoot"),
-
-  require("./generated/MutationObserver"),
-  require("./generated/MutationRecord"),
-
-  require("./generated/Headers"),
-  require("./generated/AbortController"),
-  require("./generated/AbortSignal")
-];
+const generatedInterfaces = {
```

Comment:
You could probably leave this as-is and then do a single pass over the array to build a hash table from `arrayMember.interface.name` or something like that... seems a bit nicer for maintenance.

Reply:
The `interface` is not exposed anymore on the generated interfaces since the constructor and prototype reform. And there is, unfortunately, no other way today to retrieve the interface name.

Reply:
Good point. Can we add a comment explaining that this needs to stay as-is in order to make bundlers not freak out? Right now it seems like a really tempting refactoring target.